apps/core/src/utils/nftStandard.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+export type LinkData = {
+    href: string;
+    display: string;
+};
+
+function toLinkData(link: string): LinkData | string | null {
+    try {
+        const url = new URL(link);
+        return { href: link, display: url.hostname };
+    } catch (e) {
+        return link || null;
+    }
+}
+
+export function processDisplay(display: Record<string, string> | null) {
```

Comment:
We shouldn't use this kind of pattern, given display data can include anything outside of these fields that we'll still display in some contexts. Would rather just handle the display fields piecemeal. This also aligns with our migration away from view-model-like processing of data (which this is closer to).